doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -284,155 +294,81 @@ Induction
-   :n:`(p`:sub:`1` :n:`, ... , p`:sub:`n` :n:`)` can be used instead of
-   :n:`[ p`:sub:`1` :n:`... p`:sub:`n` :n:`]`.
+   .. exn:: Not an inductive product.
+      :undocumented:
 
-.. tacv:: induction @term with @bindings
+   .. exn:: Unable to find an instance for the variables @ident … @ident.
 
-   This behaves like :tacn:`induction` providing explicit instances for the
-   premises of the type of :n:`term` (see :ref:`bindings`).
+      Use the variant :tacn:`elim` `… with …` instead.
 
-.. tacv:: einduction @term
-   :name: einduction
+.. tacn:: einduction {+, @induction_clause } {? @induction_principle }
 
-   This tactic behaves like :tacn:`induction` except that it does not fail if
-   some dependent premise of the type of :n:`@term` is not inferable. Instead,
+   Behaves like :tacn:`induction` except that it does not fail if
+   some dependent premise of the type of :n:`@one_term` is not inferrable. Instead,
    the unresolved premises are posed as existential variables to be inferred
    later, in the same way as :tacn:`eapply` does.
 
-.. tacv:: induction @term using @term
-   :name: induction … using …
-
-   This behaves as :tacn:`induction`  but using :n:`@term` as induction scheme.
-   It does not expect the conclusion of the type of the first :n:`@term` to be
-   inductive.
-
-.. tacv:: induction @term using @term with @bindings
-
-   This behaves as :tacn:`induction … using …` but also providing instances
-   for the premises of the type of the second :n:`@term`.
-
-.. tacv:: induction {+, @term} using @qualid
-
-   This syntax is used for the case :n:`@qualid` denotes an induction principle
-   with complex predicates as the induction principles generated by
-   ``Function`` or ``Functional Scheme`` may be.
-
-.. tacv:: induction @term in @goal_occurrences
-
-   This syntax is used for selecting which occurrences of :n:`@term` the
-   induction has to be carried on. The :n:`in @goal_occurrences` clause is an
-   occurrence clause whose syntax and behavior is described in
-   :ref:`occurrences sets <occurrenceclauses>`. If variables or hypotheses not
-   mentioning :n:`@term` in their type are listed in :n:`@goal_occurrences`,
-   those are generalized as well in the statement to prove.
-
-   .. example::
-
-      .. coqtop:: reset all
-
-         Lemma comm x y : x + y = y + x.
-         induction y in x |-   *.
-         Show 2.
+.. tacn:: elim @one_term_with_bindings {? using @one_term {? with @bindings } }
 
-.. tacv:: induction @term with @bindings as @or_and_intropattern using @term with @bindings in @goal_occurrences
-          einduction @term with @bindings as @or_and_intropattern using @term with @bindings in @goal_occurrences
+   An older, more basic induction tactic.  We recommend using :tacn:`induction`
+   instead where possible.  :n:`@one_term` must have an inductive
+   type.  The tactic applies the appropriate :term:`induction principle` based
```

Comment:
The fact that the tactic selects the "appropriate" induction principle (among the various ones that are automatically generated) when the `@induction_principle` clause is not provided is something else that could be added to the description of `induction`.

Reply:
Done.  Would it be better or worse to say "based on the _sort_ of the goal"?

Reply:
It would be better indeed.